### DISCUSSION TITLE
how to use the $EDITOR as the default rename tool?

### DISCUSSION THREAD

**Discussion by OriLiMu** [sxyazi/yazi#1254] [1. Q&A] [answered]
I don't want to use The default input box. I want nvim as the default tool to rename. But after I read the doc, there seems no options for it. SO any way to make it?

**Comment by AnirudhG07** [comment]
You can do that by pressing `<Space>` and `r` consecutively. Yazi takes the selected files(not hovered) and bulk renames it in neovim(EDITOR).
So selecting one file and pressing "r" opens neovim

**Reply by OriLiMu** [reply]
I don't want press two keys. I  want it replaced original r key

**Reply by AnirudhG07** [reply, marked as answer]
```
[[manager.prepend_keymap]]
on  = [ "r"]
run = [ "select --state=none", "rename --cursor=before_ext"]
desc  = "rename in neovim"
```
Here you go.

**Reply by AnirudhG07** [reply]
but this has will ask for a confirmation like bulk renaming. sooo with this you ll have to deal with that.

**Reply by OriLiMu** [reply]
> ```
> [[manager.prepend_keymap]]
> on  = [ "r"]
> run = [ "select --state=none", "rename --cursor=before_ext"]
> desc  = "rename in neovim"
> ```
> 
> Here you go.

After quit the $Editor, there is a confirm dialog, is there a way to disable it? 

I try the `--force` option not work

	`{ on = [ "r" ],         run = ["select --state=none", "rename --cursor=start --force"], desc = "Rename in neovim" },`

**Reply by AnirudhG07** [reply]
Yes it's not working for me either. Please make a new discussion question on this. 
Also -
```
`{ on = [ "r" ],         run = ["select --state=none", "rename --cursor=start --force", "select --state=none"], desc = "Rename in neovim" },`
```
This will deselect the file after you rename it. Else you would see it selected(atleast I did). If it already is deselected for you, the file, then its fine
